File: scratch/make_user_premium.js
const fs = require('fs');
const path = require('path');
const admin = require('firebase-admin');

// Load .env.local
const dotenv = require('dotenv');
const envPath = path.join(__dirname, '..', '.env.local');
if (fs.existsSync(envPath)) {
  dotenv.config({ path: envPath });
}

const firebaseServiceAccountStr = process.env.FIREBASE_SERVICE_ACCOUNT_JSON;
if (!firebaseServiceAccountStr) {
  console.error("FIREBASE_SERVICE_ACCOUNT_JSON not found in environment!");
  process.exit(1);
}

const serviceAccount = JSON.parse(firebaseServiceAccountStr);

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

const db = admin.firestore();
const email = "ueservicesllc1@gmail.com";
const password = "PremiumPassword123!";

async function run() {
  let user;
  try {
    user = await admin.auth().getUserByEmail(email);
    console.log(`User ${email} already exists with UID: ${user.uid}`);
  } catch (error) {
    if (error.code === 'auth/user-not-found') {
      user = await admin.auth().createUser({
        email: email,
        password: password,
        emailVerified: true
      });
      console.log(`Created new user ${email} with UID: ${user.uid} and default password: ${password}`);
    } else {
      throw error;
    }
  }

  // Update/Set document in users collection
  const userRef = db.collection('users').doc(user.uid);
  await userRef.set({
    planId: 'ultra',
    tokenBalance: 20000,
    freeSeparationUsed: false,
    email: email
  }, { merge: true });

  console.log(`Successfully configured user ${email} (UID: ${user.uid}) as Premium (planId: ultra, 20000 tokens)`);
}

run().catch(err => {
  console.error("Error setting user premium:", err);
  process.exit(1);
});

File: scratch/make_user_premium.py
import os
import json
import firebase_admin
from firebase_admin import credentials, auth, firestore

def load_env_local(filepath):
    if not os.path.exists(filepath):
        return
    with open(filepath, "r", encoding="utf-8") as f:
        for line in f:
            line = line.strip()
            if not line or line.startswith("#"):
                continue
            if "=" in line:
                k, v = line.split("=", 1)
                os.environ[k.strip()] = v.strip()

# Load local env vars
load_env_local("e:/moises2/.env.local")

# Get firebase credentials
firebase_sa_json = os.getenv("FIREBASE_SERVICE_ACCOUNT_JSON", "")
if not firebase_sa_json:
    raise ValueError("FIREBASE_SERVICE_ACCOUNT_JSON not found in environment!")

cred = credentials.Certificate(json.loads(firebase_sa_json))
firebase_admin.initialize_app(cred)

db = firestore.client()

email = "ueservicesllc1@gmail.com"
password = "PremiumPassword123!"

try:
    # Try fetching existing user
    user = auth.get_user_by_email(email)
    print(f"User {email} exists with UID: {user.uid}")
except auth.UserNotFoundError:
    # Create user
    user = auth.create_user(
        email=email,
        password=password,
        email_verified=True
    )
    print(f"Created new user {email} with UID: {user.uid} and password: {password}")

# Now update the user document in Firestore to premium
# Let's set the plan to ultra (20000 tokens)
uid = user.uid
user_ref = db.collection("users").document(uid)
user_ref.set({
    "planId": "ultra",
    "tokenBalance": 20000,
    "freeSeparationUsed": False,
    "email": email
}, merge=True)

print(f"Successfully configured user {email} (UID: {uid}) as Premium (planId: ultra, 20000 tokens)")
